neuraxle/steps/loop.py

```diff
@@ -157,38 +157,19 @@ def __init__(self, wrapped: BaseStep, copy_op=copy.deepcopy, cache_folder_when_n
         self.steps_as_tuple: List[NamedTupleList] = []
         self.copy_op = copy_op
 
-    def set_train(self, is_train: bool = True):
-        MetaStepMixin.set_train(self, is_train)
-        [step.set_train(is_train) for _, step in self]
-        return self
-
-    def set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
-        MetaStepMixin.set_hyperparams(self, hyperparams)
-        self.steps_as_tuple = [(name, step.set_hyperparams(self.wrapped.get_hyperparams())) for name, step in self]
-        return self
-
-    def update_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
-        """
-        Update the step hyperparameters without removing the already-set hyperparameters.
-        Please refer to :func:`~BaseStep.update_hyperparams`.
+    def apply(self, method_name: str, step_name=None, *kargs, **kwargs) -> Dict:
+        results = MetaStepMixin.apply(self, method_name=method_name, step_name=step_name, *kargs, **kwargs)
+        if len(self) > 0:
+            for _, step in self:
+                results.update(step.apply(method_name=method_name, step_name=step_name, *kargs, **kwargs))
```

Comment:
Why do you need to use the `MetaStepMixin.apply` here before the `step.apply`? Is it because the self.wrapped is different from all the cloned steps? Comments in the code would probably help here.